acceptance_test.go

```diff
@@ -0,0 +1,175 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package vitess implements Vitess connector for Conduit.
+// It provides both, a source and a destination Vitess connector.
+
+package vitess
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/brianvoe/gofakeit"
+	"github.com/conduitio-labs/conduit-connector-vitess/config"
+	"github.com/conduitio-labs/conduit-connector-vitess/source"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/matryer/is"
+	"go.uber.org/goleak"
+	"vitess.io/vitess/go/vt/vitessdriver"
+)
+
+var (
+	queryCreateTestTable  = `create table %s (id int, name text, primary key(id));`
+	queryCreateTestVindex = `alter vschema on %s add vindex hash(id) using hash;`
+	queryDropTestTable    = `drop table %s;`
+
+	metadataAction = "action"
+	actionInsert   = "insert"
+)
+
+type driver struct {
+	sdk.ConfigurableAcceptanceTestDriver
+
+	counter int64
+}
+
+// GenerateRecord generates a random sdk.Record.
+func (d *driver) GenerateRecord(t *testing.T) sdk.Record {
+	atomic.AddInt64(&d.counter, 1)
+
+	return sdk.Record{
+		Position: nil,
+		Metadata: map[string]string{
+			metadataAction:  actionInsert,
+			config.KeyTable: d.Config.DestinationConfig[config.KeyTable],
+		},
+		Key: sdk.StructuredData{
+			"id": d.counter,
+		},
+		Payload: sdk.RawData(
+			fmt.Sprintf(
+				`{"id":%d,"name":"%s"}`, d.counter, gofakeit.Name(),
+			),
+		),
+	}
+}
+
+//nolint:paralleltest // we don't need paralleltest for the Acceptance tests.
+func TestAcceptance(t *testing.T) {
+	cfg := prepareConfig(t)
+
+	sdk.AcceptanceTest(t, &driver{
+		ConfigurableAcceptanceTestDriver: sdk.ConfigurableAcceptanceTestDriver{
+			Config: sdk.ConfigurableAcceptanceTestDriverConfig{
+				Connector:         Connector,
+				SourceConfig:      cfg,
+				DestinationConfig: cfg,
+				BeforeTest:        beforeTest(t, cfg),
+				GoleakOptions: []goleak.Option{
+					// these leaks spawn Vitess libraries, there's no way to stop them manually
```

Comment:
could you please create issue for vittes library. It is only if you want, you can close this comment

Reply:
It will take some time to investigate from where exactly these problems come within the Vitess libraries, so I will close the comment for now